dub.sdl

```diff
@@ -0,0 +1,63 @@
+name "dmd"
+description "The DMD compiler"
+authors "Walter Bright"
+copyright "Copyright © 1999-2017, Digital Mars"
+license "BSL-1.0"
+
+targetType "library"
+
+// This is required because otherwise Dub will include all source files in src.
+// Just picked an existing directory which doesn't contain any D source files.
+sourcePaths "docs"
+
+// This is required because Dub needs to have at lease one source file
+sourceFiles "src/ddmd/package.d"
+
+dependency ":parser" version="*"
+
+subPackage {
+  name "idgen"
+  targetType "executable"
+  sourcePaths "docs"
+  sourceFiles "src/ddmd/idgen.d"
+}
+
+subPackage {
+  name "root"
+  targetType "library"
+  sourcePaths "src/ddmd/root"
+}
+
+subPackage {
+  name "lexer"
+  targetType "library"
+  sourcePaths "docs"
+
+  sourceFiles \
+    "src/ddmd/entity.d" \
+    "src/ddmd/errors.d" \
+    "src/ddmd/globals.d" \
+    "src/ddmd/id.d" \
+    "src/ddmd/identifier.d" \
+    "src/ddmd/lexer.d" \
+    "src/ddmd/tokens.d" \
+    "src/ddmd/utf.d"
+
+  // stringImportPaths cannot be used because it will make Dub extremely slow
+  dflags "-J$PACKAGE_DIR"
+
+  preBuildCommands "dub run dmd:idgen -- $PACKAGE_DIR"
+  dependency "dmd:root" version="*"
+}
+
+subPackage {
+  name "parser"
+  targetType "library"
+  sourcePaths "docs"
```

Comment:
`sourcePaths "docs"` can be replaced by an empty `sourcePaths`. Defining an empty list of source paths is enough to inhibit the auto-detection of the "src" folder.